Arbres/classe_noeud.py

```diff
@@ -1,27 +1,49 @@
 class Noeud:
-    # attribut de classe
+    #attribut de classe
     arbre_vide = None
 
+    """Co nstruct eu r d'un no eud d'u n arbre bin aire"""
     def __init__(self, v, g, d):
         self._valeur = v
         self._gauche = g
         self._droit = d
-    
+
+    #selecteurs
     def valeur(self):
         return self._valeur
     
     def gauche(self):
         return self._gauche
-    
+
     def droit(self):
         return self._droit
-    
-    def est_vide(arbre):
+
+    #methode de classe
+    def est_vide (arbre):
         return arbre is Noeud.arbre_vide
+    
+    def est_feuille(arbre):
+        if arbre.gauche() == Noeud.arbre_vide and arbre.droit() == Noeud.arbre_vide:
+            return True
+        return False
+    
+def compte_feuilles(arbre : Noeud):
+    if arbre == Noeud.arbre_vide:
+        return 0
+    if arbre.est_feuille():
+        return 1
+    return compte_feuilles(arbre.gauche()) + compte_feuilles(arbre.droit())
 
```

Comment:
c'est pas plutôt dans la class ?


```suggestion
    def compte_feuilles(arbre: Noeud):
        if arbre == Noeud.arbre_vide:
            return 0
        if arbre.est_feuille():
            return 1
        return compte_feuilles(arbre.gauche()) + compte_feuilles(arbre.droit())

```